### ISSUE TITLE
Different result-set between with table alias and without on LEFT JOIN ... WHERE ...

### ISSUE THREAD

**Issue by SunguckLee** [yandex/ClickHouse#3979]
Usually, in SQL below two queries produce same results.
```sql
SELECT * FROM (SELECT * FROM tab WHERE fd1=?) sub;
     vs 
SELECT * FROM (SELECT t1.* FROM tab t1. WHERE t1.fd1=?) sub;
```

But on ClickHouse, result sets are different.
```sql
-- // Prepare test table and data
CREATE TABLE table1 ( 
  user_id String,  
  a_type String,  
  a_amount Int64,
  created_at DateTime
) ENGINE = MergeTree PARTITION BY toYYYYMM(created_at) ORDER BY (a_type, created_at) SETTINGS index_granularity = 8192;

CREATE TABLE table2 ( 
  user_id String,  
  created_at DateTime
) ENGINE = MergeTree PARTITION BY toYYYYMM(created_at) ORDER BY (created_at) SETTINGS index_granularity = 8192;

INSERT INTO table1 VALUES ('matt', 'A', 10, '2018-01-01 00:00:00'),('lara', 'A', 10, '2018-01-01 00:00:00');
INSERT INTO table2 VALUES ('matt', '2018-01-01 00:00:00');

-- // Without table alias in subquery
SELECT t1.user_id, t2.user_id
FROM
  (SELECT user_id FROM table1 WHERE created_at<'2019-01-01 00:00:00' AND a_type='A') t1
  ANY LEFT JOIN
  (SELECT user_id FROM table2 WHERE created_at<'2019-01-01 00:00:00') t2 USING (user_id)
WHERE empty(t2.user_id)=1

┌─user_id─┬─t2.user_id─┐
│ matt    │            │
│ lara    │            │
└─────────┴────────────┘

-- // With table alias in subquery
SELECT t1.user_id, t2.user_id
FROM
  (SELECT it1.user_id FROM table1 it1 WHERE it1.created_at<'2019-01-01 00:00:00' AND it1.a_type='A') t1
  ANY LEFT JOIN
  (SELECT it2.user_id FROM table2 it2 WHERE it2.created_at<'2019-01-01 00:00:00') t2 USING (user_id)
WHERE empty(t2.user_id)=1

┌─user_id─┬─t2.user_id─┐
│ lara    │            │
└─────────┴────────────┘
```

Looks like WHERE conditions of outer-query is pushed down to subquery inadequately.

**Comment by SunguckLee** [comment]
```
SELECT version()
┌─version()─┐
│ 18.14.8   │
└───────────┘
```

**Comment by den-crane** [comment]
fixed in 18.16.0 https://github.com/yandex/ClickHouse/pull/3588

both queries return


lara |  
-- | --

18.16.0

Workaround: set enable_optimize_predicate_expression = 0

**Comment by SunguckLee** [comment]
I will upgrade & check it again.
Thanks.
